server.js

```diff
@@ -21,7 +16,72 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = expressListEndpoints(app);
+
+  res.json(endpoints);
+});
+
+// Filter books
+const filterBooks = (booksData, query) => {
+  let filteredBooks = [...booksData];
+
+  // Filter by title
+  if (query.title) {
+    const titleRegex = new RegExp(query.title, "i");
+    filteredBooks = filteredBooks.filter((book) => titleRegex.test(book.title));
+  }
+
+  // Filter by author
+  if (query.authors) {
+    const authorsRegex = new RegExp(query.authors, "i");
+    filteredBooks = filteredBooks.filter((book) =>
+      authorsRegex.test(book.authors)
+    );
+  }
+
+  // Filter by minimum rating
+  if (query.minRating) {
+    const minRating = parseFloat(query.minRating);
+    filteredBooks = filteredBooks.filter(
+      (book) => book.average_rating >= minRating
+    );
+  }
+  // Response if no books are found
+  if (filteredBooks.length === 0) {
+    return "Not found";
+  }
+
+  return filteredBooks;
+};
```

Comment:
nice you added some queries to filter the data, make sure to add it to the documented endpoints you listed with `expressListEndpoints`, for example:

```javascript
const generateQueryUsage = (endpoint) => {
  const queryUsage = {};
  if (endpoint.path === '/books') {
    queryUsage.title = 'Filter by book title (case-insensitive)';
    queryUsage.authors = 'Filter by author name (case-insensitive)';
    queryUsage.minRating = 'Filter by minimum average rating';
  }
  return queryUsage;
};

app.get("/", (req, res) => {
  const endpoints = expressListEndpoints(app).map((endpoint) => ({
    path: endpoint.path,
    methods: endpoint.methods,
    query: generateQueryUsage(endpoint),
  }));
  res.json(endpoints);
});
```